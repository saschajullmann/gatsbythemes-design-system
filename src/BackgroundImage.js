import styled from '@emotion/styled'
import PropTypes from 'prop-types'
import { width } from 'styled-system'
import theme from './theme'

const image = props =>
  props.image ? { backgroundImage: `url(${props.image})` } : null

const height = props => (props.height ? { height: props.height } : null)

const BackgroundImage = styled.div`
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
  background-color: ${props => props.theme.colors.gray};
  ${image} ${height} ${width};
`
const numberStringOrArray = PropTypes.oneOfType([
  PropTypes.number,
  PropTypes.string,
  PropTypes.array
])

BackgroundImage.propTypes = {
  /** background-image url */
  image: PropTypes.string,
  width: numberStringOrArray
}

BackgroundImage.defaultProps = {
  theme: theme
}

BackgroundImage.displayName = 'BackgroundImage'

export default BackgroundImage
