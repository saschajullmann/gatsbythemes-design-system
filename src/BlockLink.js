import styled from '@emotion/styled'
import Link from './Link'

const BlockLink = styled(Link)`
  display: block;
  color: inherit;
  text-decoration: none;
`

BlockLink.displayName = 'BlockLink'

export default BlockLink
