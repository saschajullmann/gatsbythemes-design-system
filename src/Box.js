import styled from '@emotion/styled'
import { space, width, color, textAlign } from 'styled-system'
import PropTypes from 'prop-types'
import theme from './theme'

const Box = styled.div`
  ${space} ${width} ${color} ${textAlign}
`

Box.displayName = 'Box'

Box.defaultProps = {
  theme: theme
}

Box.propTypes = {
  ...space.propTypes,
  ...width.propTypes,
  ...color.propTypes,
  ...textAlign.propTypes
}

export default Box
