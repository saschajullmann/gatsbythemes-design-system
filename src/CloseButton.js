import React from 'react'
import IconButton from './IconButton'
import PropTypes from 'prop-types'
import theme from './theme'

const CloseButton = props => <IconButton {...props} name="close" />

CloseButton.defaultProps = {
  size: 24,
  title: 'close'
}

CloseButton.propTypes = {
  onClick: PropTypes.func,
  size: PropTypes.number,
  title: PropTypes.string
}

CloseButton.displayName = 'CloseButton'

export default CloseButton
