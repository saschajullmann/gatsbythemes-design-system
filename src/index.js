export { default as Absolute } from './Absolute'
export { default as BackgroundImage } from './BackgroundImage'
export { default as Badge } from './Badge'
export { default as Banner } from './Banner'
export { default as BlockLink } from './BlockLink'
export { default as Box } from './Box'
export { default as Button } from './Button'
export { default as Card } from './Card'
export { default as CloseButton } from './CloseButton'
export { default as Container } from './Container'
export { default as Divider } from './Divider'
export { default as Flag } from './Flag'
export { default as Flex } from './Flex'
export { default as GreenButton } from './GreenButton'
export { default as Heading } from './Heading'
export { default as Hide } from './Hide'
export { default as Hug } from './Hug'
export { default as Icon } from './Icon'
export { default as IconButton } from './IconButton'
export { default as IconField } from './IconField'
export { default as Image } from './Image'
export { default as Input } from './Input'
export { default as InputField } from './FormField'
export { default as FormField } from './FormField'
export { default as InputGroup } from './InputGroup'
export { default as Label } from './Label'
export { default as Link } from './Link'
export { default as OutlineButton } from './OutlineButton'
export { default as Radio } from './Radio'
export { default as RatingBadge } from './RatingBadge'
export { default as RedButton } from './RedButton'
export { default as Relative } from './Relative'
export { default as Stamp } from './Stamp'
export { default as Select } from './Select'
export { default as Text } from './Text'
export { default as Tooltip } from './Tooltip'
export { default as ToggleBadge } from './ToggleBadge'
export { default as Truncate } from './Truncate'
export { default as Checkbox } from './Checkbox'

export { default as theme } from './theme'
export { default as ThemeProvider } from './ThemeProvider'

export * from './theme'