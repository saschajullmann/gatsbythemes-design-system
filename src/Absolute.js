import styled from '@emotion/styled'
import Box from './Box'
import { top, right, bottom, left, zIndex } from 'styled-system'

const Absolute = styled(Box)`
  position: absolute;
  ${top} ${bottom} ${left} ${right}
  ${zIndex}
`

Absolute.propTypes = {
  ...top.propTypes,
  ...right.propTypes,
  ...bottom.propTypes,
  ...left.propTypes,
  ...zIndex.propTypes
}

Absolute.displayName = 'Absolute'

export default Absolute
