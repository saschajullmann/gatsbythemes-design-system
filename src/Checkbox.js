import React from 'react'
import styled from '@emotion/styled'
import { css } from '@emotion/core'
import PropTypes from 'prop-types'
import theme from './theme'
import Icon from './Icon'
import Box from './Box'

const Checkbox = props => {
  const { disabled, size } = props
  return (
    <CheckBoxWrapper disabled={disabled}>
      <StyledInput type="checkbox" {...props} />
      <Icon name="boxChecked" size={size} data-name="checked" />
      <Icon name="boxEmpty" size={size} data-name="empty" />
    </CheckBoxWrapper>
  )
}

const CheckBoxWrapper = styled(Box)`
  display: inline-block;
  position: relative;
  vertical-align: middle;
  cursor: pointer;
  color: ${props =>
    props.disabled ? props.theme.colors.borderGray : props.theme.colors.gray};

  svg[data-name='checked'] {
    display: none;
  }

  > input:checked {
    & ~ svg[data-name='checked'] {
      display: inline-block;
      color: ${props =>
    props.disabled
      ? props.theme.colors.borderGray
      : props.theme.colors.blue};
    }

    & ~ svg[data-name='empty'] {
      display: none;
    }
  }
`

CheckBoxWrapper.defaultProps = {
  theme
}

const StyledInput = styled.input`
  appearance: none;
  opacity: 0;
  position: absolute;
  z-index: 0;
`

Checkbox.displayName = 'Checkbox'

Checkbox.propTypes = {
  id: PropTypes.string.isRequired,
  size: PropTypes.number,
  onChange: PropTypes.func.isRequired
}

Checkbox.defaultProps = {
  size: 20
}

export default Checkbox
