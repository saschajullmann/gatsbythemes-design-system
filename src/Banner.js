import React from 'react'
import Box from './Box'
import Flex from './Flex'
import Text from './Text'
import Icon from './Icon'
import CloseButton from './CloseButton'
import Heading from './Heading'
import PropTypes from 'prop-types'

const bannerColors = {
  green: {
    backgroundColor: 'green',
    color: 'white',
    icon: 'success'
  },
  lightGreen: {
    backgroundColor: 'lightGreen',
    color: 'darkGreen',
    icon: 'success'
  },
  red: {
    backgroundColor: 'red',
    color: 'white',
    icon: 'warning'
  },
  lightRed: {
    backgroundColor: 'lightRed',
    color: 'darkRed',
    icon: 'warning'
  },
  orange: {
    backgroundColor: 'orange',
    color: 'white',
    icon: 'attention'
  },
  blue: {
    backgroundColor: 'blue',
    color: 'white',
    icon: 'information'
  },
  lightBlue: {
    backgroundColor: 'lightBlue',
    color: 'darkBlue',
    icon: 'information'
  }
}

const Banner = props => {
  const bannerColor = bannerColors[props.bg] || {}
  const icon = props.iconName || bannerColor.icon

  return (
    <Box
      {...props}
      bg={bannerColor.backgroundColor || props.bg}
      color={bannerColor.color || props.color}
    >
      <Flex justifyContent="space-between" alignItems="flex-start">
        {!!icon && !!props.showIcon && (
          <Icon name={icon} mr={2} size={24} mt="-2px" />
        )}
        <Box w={1}>
          <Text textAlign={props.textAlign}>
            <Heading.h5>{props.header}</Heading.h5>
            <Text.span fontSize={1}>{props.text}</Text.span>
            {props.children}
          </Text>
        </Box>
        {!!props.onClose && (
          <CloseButton
            onClick={props.onClose}
            ml={2}
            size={24}
            title="close"
            mt="-2px"
          />
        )}
      </Flex>
    </Box>
  )
}

Banner.displayName = 'Banner'

Banner.propTypes = {
  header: PropTypes.string,
  iconName: PropTypes.string,
  onClose: PropTypes.func,
  showIcon: PropTypes.bool,
  text: PropTypes.node,
  textAlign: PropTypes.string
}

Banner.defaultProps = {
  bg: 'green',
  textAlign: 'left',
  showIcon: true
}

export default Banner
