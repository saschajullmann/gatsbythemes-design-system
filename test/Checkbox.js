import React from 'react'
import renderer from 'react-test-renderer'
import { Checkbox, Icon, theme } from '../src'
import { render, fireEvent, cleanup } from 'react-testing-library'

afterEach(cleanup)

describe('Checkbox', () => {
  test('renders without the theme passed specifically', () => {
    const onChange = jest.fn()
    const json = renderer
      .create(<Checkbox id="check-box" onChange={onChange} />)
      .toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with the theme passed specifically', () => {
    const json = renderer
      .create(<Checkbox id="check-box" theme={theme} />)
      .toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders checked when defaultChecked prop is passed as true', () => {
    const json = renderer
      .create(<Checkbox id="check-box" defaultChecked />)
      .toJSON()
    const [input] = json.children
    expect(input.props.defaultChecked).toBe(true)
  })

  test('renders disabled with disabled prop', () => {
    const onChange = jest.fn()
    const json = renderer
      .create(<Checkbox id="check-box" disabled={true} onChange={onChange} />)
      .toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders disabled with defaultChecked', () => {
    const onChange = jest.fn()
    const json = renderer
      .create(
        <Checkbox
          id="check-box"
          disabled={true}
          defaultChecked={true}
          onChange={onChange}
        />
      )
      .toJSON()
    expect(json).toMatchSnapshot()
  })

  test('calls onChange when clicked', () => {
    const onChange = jest.fn()
    const { container } = render(
      <Checkbox id="check-box" onChange={onChange} />
    )
    const checkbox = container.querySelector('[type=checkbox]')
    fireEvent.click(checkbox)
    expect(onChange).toHaveBeenCalled()
  })
})
