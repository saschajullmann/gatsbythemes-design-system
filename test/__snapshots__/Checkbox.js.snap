// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders disabled with defaultChecked 1`] = `
.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c0 {
  display: inline-block;
  position: relative;
  vertical-align: middle;
  cursor: pointer;
  color: #c0cad5;
}

.c0 svg[data-name='checked'] {
  display: none;
}

.c0 > input:checked ~ svg[data-name='checked'] {
  display: inline-block;
  color: #c0cad5;
}

.c0 > input:checked ~ svg[data-name='empty'] {
  display: none;
}

.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  opacity: 0;
  position: absolute;
  z-index: 0;
}

<div
  className="c0 "
  disabled={true}
>
  <input
    className="c1"
    defaultChecked={true}
    disabled={true}
    id="check-box"
    onChange={[MockFunction]}
    size={20}
    type="checkbox"
  />
  <svg
    className="c2"
    data-name="checked"
    fill="currentcolor"
    height={20}
    viewBox="0 0 24 24"
    width={20}
  >
    <path
      d="M6 3h12c1.7 0 3 1.3 3 3v12c0 1.7-1.3 3-3 3H6c-1.7 0-3-1.3-3-3V6c0-1.7 1.3-3 3-3zm4 14l9-8.6L17.6 7 10 14.3l-3.6-3.5L5 12.2l5 4.8z"
    />
  </svg>
  <svg
    className="c2"
    data-name="empty"
    fill="currentcolor"
    height={20}
    viewBox="0 0 24 24"
    width={20}
  >
    <path
      d="M6 5c-.6 0-1 .4-1 1v12c0 .6.4 1 1 1h12c.6 0 1-.4 1-1V6c0-.6-.4-1-1-1H6zm0-2h12c1.7 0 3 1.3 3 3v12c0 1.7-1.3 3-3 3H6c-1.7 0-3-1.3-3-3V6c0-1.7 1.3-3 3-3z"
    />
  </svg>
</div>
`;

exports[`Checkbox renders disabled with disabled prop 1`] = `
.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c0 {
  display: inline-block;
  position: relative;
  vertical-align: middle;
  cursor: pointer;
  color: #c0cad5;
}

.c0 svg[data-name='checked'] {
  display: none;
}

.c0 > input:checked ~ svg[data-name='checked'] {
  display: inline-block;
  color: #c0cad5;
}

.c0 > input:checked ~ svg[data-name='empty'] {
  display: none;
}

.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  opacity: 0;
  position: absolute;
  z-index: 0;
}

<div
  className="c0 "
  disabled={true}
>
  <input
    className="c1"
    disabled={true}
    id="check-box"
    onChange={[MockFunction]}
    size={20}
    type="checkbox"
  />
  <svg
    className="c2"
    data-name="checked"
    fill="currentcolor"
    height={20}
    viewBox="0 0 24 24"
    width={20}
  >
    <path
      d="M6 3h12c1.7 0 3 1.3 3 3v12c0 1.7-1.3 3-3 3H6c-1.7 0-3-1.3-3-3V6c0-1.7 1.3-3 3-3zm4 14l9-8.6L17.6 7 10 14.3l-3.6-3.5L5 12.2l5 4.8z"
    />
  </svg>
  <svg
    className="c2"
    data-name="empty"
    fill="currentcolor"
    height={20}
    viewBox="0 0 24 24"
    width={20}
  >
    <path
      d="M6 5c-.6 0-1 .4-1 1v12c0 .6.4 1 1 1h12c.6 0 1-.4 1-1V6c0-.6-.4-1-1-1H6zm0-2h12c1.7 0 3 1.3 3 3v12c0 1.7-1.3 3-3 3H6c-1.7 0-3-1.3-3-3V6c0-1.7 1.3-3 3-3z"
    />
  </svg>
</div>
`;

exports[`Checkbox renders with the theme passed specifically 1`] = `
.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c0 {
  display: inline-block;
  position: relative;
  vertical-align: middle;
  cursor: pointer;
  color: #4f6f8f;
}

.c0 svg[data-name='checked'] {
  display: none;
}

.c0 > input:checked ~ svg[data-name='checked'] {
  display: inline-block;
  color: #007aff;
}

.c0 > input:checked ~ svg[data-name='empty'] {
  display: none;
}

.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  opacity: 0;
  position: absolute;
  z-index: 0;
}

<div
  className="c0 "
>
  <input
    className="c1"
    id="check-box"
    size={20}
    type="checkbox"
  />
  <svg
    className="c2"
    data-name="checked"
    fill="currentcolor"
    height={20}
    viewBox="0 0 24 24"
    width={20}
  >
    <path
      d="M6 3h12c1.7 0 3 1.3 3 3v12c0 1.7-1.3 3-3 3H6c-1.7 0-3-1.3-3-3V6c0-1.7 1.3-3 3-3zm4 14l9-8.6L17.6 7 10 14.3l-3.6-3.5L5 12.2l5 4.8z"
    />
  </svg>
  <svg
    className="c2"
    data-name="empty"
    fill="currentcolor"
    height={20}
    viewBox="0 0 24 24"
    width={20}
  >
    <path
      d="M6 5c-.6 0-1 .4-1 1v12c0 .6.4 1 1 1h12c.6 0 1-.4 1-1V6c0-.6-.4-1-1-1H6zm0-2h12c1.7 0 3 1.3 3 3v12c0 1.7-1.3 3-3 3H6c-1.7 0-3-1.3-3-3V6c0-1.7 1.3-3 3-3z"
    />
  </svg>
</div>
`;

exports[`Checkbox renders without the theme passed specifically 1`] = `
.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c0 {
  display: inline-block;
  position: relative;
  vertical-align: middle;
  cursor: pointer;
  color: #4f6f8f;
}

.c0 svg[data-name='checked'] {
  display: none;
}

.c0 > input:checked ~ svg[data-name='checked'] {
  display: inline-block;
  color: #007aff;
}

.c0 > input:checked ~ svg[data-name='empty'] {
  display: none;
}

.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  opacity: 0;
  position: absolute;
  z-index: 0;
}

<div
  className="c0 "
>
  <input
    className="c1"
    id="check-box"
    onChange={[MockFunction]}
    size={20}
    type="checkbox"
  />
  <svg
    className="c2"
    data-name="checked"
    fill="currentcolor"
    height={20}
    viewBox="0 0 24 24"
    width={20}
  >
    <path
      d="M6 3h12c1.7 0 3 1.3 3 3v12c0 1.7-1.3 3-3 3H6c-1.7 0-3-1.3-3-3V6c0-1.7 1.3-3 3-3zm4 14l9-8.6L17.6 7 10 14.3l-3.6-3.5L5 12.2l5 4.8z"
    />
  </svg>
  <svg
    className="c2"
    data-name="empty"
    fill="currentcolor"
    height={20}
    viewBox="0 0 24 24"
    width={20}
  >
    <path
      d="M6 5c-.6 0-1 .4-1 1v12c0 .6.4 1 1 1h12c.6 0 1-.4 1-1V6c0-.6-.4-1-1-1H6zm0-2h12c1.7 0 3 1.3 3 3v12c0 1.7-1.3 3-3 3H6c-1.7 0-3-1.3-3-3V6c0-1.7 1.3-3 3-3z"
    />
  </svg>
</div>
`;
